[export] Update docs

yiming0416 opened the pull request:
Summary:
Update export docs. Including:
1. Update the output graph.
2. Misc fixes for examples.

Test Plan: CI

Differential Revision: D66726729

angelayi commented on docs/source/export.rst:
```diff
@@ -39,28 +39,27 @@ serialized.
+            def forward(self, x: "f32[10, 10]", y: "f32[10, 10]"):
                 # code: a = torch.sin(x)
-                sin: f32[10, 10] = torch.ops.aten.sin.default(arg0_1);
+                sin: "f32[10, 10]" = torch.ops.aten.sin.default(x);  x = None
```
nit: I would prefer to remove the stuff after the semicolon :P

yiming0416 replied:
I can do that. I kept them here because I saw in the section of `Export for Training and Inference` we have kept the contents after the semicolon. Shall we also remove those?

Also, would users be confused if they see the print-out contents are not exactly matching the doc here?

angelayi commented on docs/source/export.ir_spec.rst:
```diff
@@ -102,16 +102,25 @@ Graph of GraphModule. The `output` node of a Graph represents the **outputs**
 of the Graph of GraphModule.
 
 Example::
```
you might need to add an extra space after this

angelayi reviewed (commented):
Let's wait for the docs to be generated from your PR before stamping (https://docs-preview.pytorch.org/pytorch/pytorch/142011/index.html)

angelayi commented on docs/source/export.rst:
```diff
@@ -39,28 +39,27 @@ serialized.
 
     ExportedProgram:
         class GraphModule(torch.nn.Module):
```
can you also update [this line](https://github.com/pytorch/pytorch/blob/3be26ae2af59a598306e0654235e22304df89842/docs/source/export.rst?plain=1#L401-L402) to use  2 backticks instead of 1? \`\`conv2d\`\` instead of \`conv2d\`

yiming0416 committed:
[export] Update docs (#142011)

Summary:

Update export docs. Including:
1. Update the output graph.
2. Misc fixes for examples.

Test Plan: CI

Differential Revision: D66726729
